core/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -343,38 +331,35 @@ public ParsedQuery parse(XContentParser parser) {
      */
     @Nullable
     public ParsedQuery parseInnerFilter(XContentParser parser) throws IOException {
```

Comment:
Also, do you think it's feasible now to take all the parsing stuff out of this context? maybe the few components left that need this should rather take in as argument both shard context and parse context and use them properly instead of relying on shard context to parse through parse context?